### PR TITLE
Port markdown docs to rst and introduce "githublink"

### PR DESCRIPTION
**Pull request by ultmaster**
In this pull request, all the docs will be converted to rst. I will also introduce "githublink". Docs will now be automatically linked to the code/directory that is in the corresponding commit.

Still a few problems to solve:

- [x] The migration script still cannot resolve all warnings, some of which has been listed as `FIXME`.
- [x] All github links need to be rewritten.
- [x] Will upgrade sphinx to the latest version and eject recommonmark.
- [ ] I'm not sure whether there are potential risks on readthedocs, and integration with crowdin.

### CODE REVIEW

**Comment by ultmaster** [comment]
Let's do an investigation of Chinese docs through PR #3154 first.

File: docs/en_US/FeatureEngineering/GBDTSelector.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+* 
+  **X** (array-like, require) - The training input samples which shape = [n_samples, n_features]
+
+* 
+  **y** (array-like, require) - The target values (class labels in classification, real numbers in regression) which shape = [n_samples].
+
+* 
+  **lgb_params** (dict, require) - The parameters for lightgbm model. The detail you could reference `here <https://lightgbm.readthedocs.io/en/latest/Parameters.html>`__
```
Comment (liuzhe-lz) [inline docs/en_US/FeatureEngineering/GBDTSelector.rst:53]: Two underscores?

File: docs/en_US/Compression/QuickStart.rst [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+Export Compressed Model
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You can easily export the compressed model using the following API if you are pruning your model, ``state_dict`` of the sparse model weights will be stored in ``model.pth``\ , which can be loaded by ``torch.load('model.pth')``. In this exported ``model.pth``\ , the masked weights are zero.
+
+.. code-block:::: bash
+
+   pruner.export_model(model_path='model.pth')
+
+``mask_dict`` and pruned model in ``onnx`` format(\ ``input_shape`` need to be specified) can also be exported like this:
```
Comment (liuzhe-lz) [inline docs/en_US/Compression/QuickStart.rst:206]: Backslash?

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+Note: Click the **Tuner's name** to get the Tuner's installation requirements, suggested scenario, and an example configuration. A link for a detailed description of each algorithm is located at the end of the suggested scenario for each tuner. Here is an `article <../CommunitySharings/HpoComparison>`__ comparing different Tuners on several problems.
+
+Currently, we support the following algorithms:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Tuner
+     - Brief Introduction of Algorithm
+   * - `**TPE** <#TPE>`__
```
Comment (kvartet) [inline docs/en_US/Tuner/BuiltinTuner.rst:19]: There's something wrong with the bold font.

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+==============================================
+
+To fit a machine/deep learning model into different tasks/problems, hyperparameters always need to be tuned. Automating the process of hyperparaeter tuning always requires a good tuning algorithm. NNI has provided state-of-the-art tuning algorithms as part of our built-in tuners and makes them easy to use. Below is the brief summary of NNI's current built-in tuners:
+
+Note: Click the **Tuner's name** to get the Tuner's installation requirements, suggested scenario, and an example configuration. A link for a detailed description of each algorithm is located at the end of the suggested scenario for each tuner. Here is an `article <../CommunitySharings/HpoComparison>`__ comparing different Tuners on several problems.
+
+Currently, we support the following algorithms:
+
+.. list-table::
+   :header-rows: 1
```
Comment (kvartet) [inline docs/en_US/Tuner/BuiltinTuner.rst:15]: Maybe we could add `:widths: auto`

File: docs/en_US/TrialExample/Trials.rst [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+* ``@nni.report_intermediate_result``\ /\ ``@nni.report_final_result`` will send the data to assessor/tuner at that line.
+
+For more information about annotation syntax and its usage, please refer to `Annotation <../Tutorial/AnnotationSpec>`__.
+
+Step 2 - Enable NNI Annotation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the YAML configure file, you need to set *useAnnotation* to true to enable NNI annotation:
+
+.. code-block:::: bash
```
Comment (kvartet) [inline docs/en_US/TrialExample/Trials.rst:151]: It should be two colons.

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+   * - `**PPO Tuner** <#PPOTuner>`__
+     - PPO Tuner is a Reinforcement Learning tuner based on PPO algorithm. `Reference Paper <https://arxiv.org/abs/1707.06347>`__
+   * - `**PBT Tuner** <#PBTTuner>`__
+     - PBT Tuner is a simple asynchronous optimization algorithm which effectively utilizes a fixed computational budget to jointly optimize a population of models and their hyperparameters to maximize performance. `Reference Paper <https://arxiv.org/abs/1711.09846v1>`__
+
+
+Usage of Built-in Tuners
+------------------------
+
+Using a built-in tuner provided by the NNI SDK requires one to declare the  **builtinTunerName** and** classArgs** in the ``config.yml`` file. In this part, we will introduce each tuner along with information about usage and suggested scenarios, classArg requirements, and an example configuration.
```
Comment (kvartet) [inline docs/en_US/Tuner/BuiltinTuner.rst:52]: blank space

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+
+.. code-block:: yaml
+
+   # config.yml
+   tuner:
+     builtinTunerName: PBTTuner
+     classArgs:
+       optimize_mode: maximize
+
+Note that, to use this tuner, your trial code should be modified accordingly, please refer to `the document of PBTTuner <./PBTTuner>`__ for details.
```
Comment (kvartet) [inline docs/en_US/Tuner/BuiltinTuner.rst:574]: The link is broken.

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+
+PPO Tuner
+^^^^^^^^^
+
+..
+
+   Built-in Tuner Name: **PPOTuner**
+
+
+Note that the only acceptable types within the search space are ``layer_choice`` and ``input_choice``. For ``input_choice``\ , ``n_chosen`` can only be 0, 1, or [0, 1]. Note, the search space file for NAS is usually automatically generated through the command `\ ``nnictl ss_gen`` <../Tutorial/Nnictl>`__.
```
Comment (kvartet) [inline docs/en_US/Tuner/BuiltinTuner.rst:510]: the command `\ ``nnictl ss_gen`` <../Tutorial/Nnictl>` does not display properly.

**Review by QuanluZhang — APPROVED** [review]

File: docs/en_US/FeatureEngineering/GBDTSelector.rst (hunk 1 above)
Comment (ultmaster) [reply docs/en_US/FeatureEngineering/GBDTSelector.rst:53]: Use only one underscore will produce warnings. I don't fully understand this though.

File: docs/en_US/Compression/QuickStart.rst (hunk 2 above)
Comment (ultmaster) [reply docs/en_US/Compression/QuickStart.rst:206]: Blackslash is for whitespace. It's converted from markdown. Maybe we can get rid of this later.

File: docs/en_US/Compression/Overview.rst [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+Pruning algorithms compress the original network by removing redundant weights or channels of layers, which can reduce model complexity and address the over-ﬁtting issue.
+
+.. list-table::
+   :header-rows: 1
+   :widths: auto
+
+   * - Name
+     - Brief Introduction of Algorithm
+   * - `Level Pruner <https://nni.readthedocs.io/en/latest/Compression/Pruner.html#level-pruner>`__
```
Comment (kvartet) [inline docs/en_US/Compression/Overview.rst:35]: Absolute path? Not just in this doc

**Review by kvartet — APPROVED** [review]
Should we delete the folder`archive_en_US`?

**Comment by ultmaster** [comment]
> Should we delete the folder`archive_en_US`?

Let's keep it and delete it before release.

**Pull request merged**